client/app/views/steps/password.coffee

```diff
@@ -103,16 +103,19 @@ module.exports = class PasswordView extends StepView
             onboardedSteps: ['welcome', 'agreement', 'password']
         }
 
-    handleError: (error) ->
+    handleErrorMessage: (error) ->
         @error = error
         @$errorContainer.html(t(@error))
         @$errorContainer.show()
 
 
     onSubmit: (event)->
         event?.preventDefault()
-        if @passwordStrength.label is 'weak'
-            @$inputPassword.addClass('error')
-            @handleError("password too weak")
+        data = @getDataFromDOM()
+        errors = @model.validate data
+        if errors
```

Comment:
I don't understand why you duplicate the model logic into the view? (see https://github.com/cozy/cozy-proxy/blob/development/client/app/models/step.coffee#L23). Maybe is that because Error isnt returned into the view?

The answer is on `application.coffee` https://github.com/CPatchane/cozy-proxy/blob/e6a936e764995cfbb01f17d911627a9395c68552/client/app/application.coffee#L72

````
    initialize: ->
        # ...
        @onboarding.onStepFailed (step, err) => @handleStepFailed(step, err)
        

    #....

    # Update view with error message
    # only if view is still displayed
    # otherwhise dispatch the error in console
    handleStepFailed: (step, err) ->
        if @onboarding.currentStep isnt step
            console.error err.stack
        else
            @showStep step, err

`````

Does this not work?